internal/docs/design.md

```diff
@@ -108,16 +108,19 @@ https://godoc.org/github.com/google/go-cloud/blob#Bucket.NewWriter
 ## Minimize Global State
 
 As a library, the Go CDK should not introduce global state. Global state is
-difficult to reason about in large codebases, where it can be necessary for
-different parts of the application to use different states. Instead of adding
-global state, push responsibility to the application to inject the state where
-it is needed.
+[difficult to reason about in large codebases][Singletons Considered Harmful],
```

Comment:
Do we really want to link to an OOP-oriented article here? I personally find the existing text sufficient. If anything, I'd make it even shorter

Reply:
This was a to-do I had from a while back: a few folks on the team found the cited source helpful. Wire vs. non-Wire is a principle that we haven't written down anywhere else, and this seemed like the best place to put it.

Reply:
+1 since Go is not strictly object-oriented (and the first sentence of the article is "Singletons are possibly the single greatest obstacle to good object-oriented design today"). Also, wouldn't you want to discourage other types of global state (even just plain simple global variables)? A go-specific or more general anti-global-state article would be better here if we could find one...

Reply:
Reverted.